vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceOverviewManager.java

```diff
@@ -0,0 +1,298 @@
+     * @param spaceLinkMap          For holding list of all links corresponding to
+     *                              each space
+     * @param spaceToModuleLinksMap For getting list of links from one space to
+     *                              other modules
+     * @param spaceToSpaceLinksMap  For getting list of links from one space to
+     *                              other spaces
+     */
+    @Override
+    public void getSpaceLinkMap(String contextPath, ObjectMapper mapper, ArrayNode nodeArray,
+            Map<String, List<String>> spaceLinkMap, Map<String, List<ModuleLinkDisplay>> spaceToModuleLinksMap,
+            Map<String, List<SpaceLinkDisplay>> spaceToSpaceLinksMap) {
+
+        Iterable<Space> allSpacesList = spaceRepo.findAll();
+        if (allSpacesList != null) {
+            createSpaceNodeInOverviewGraph(contextPath, mapper, nodeArray);
+            for (Space space : allSpacesList) {
+                List<ModuleLinkDisplay> spaceToModulelinksList = spaceToModuleLinksMap.get(space.getId());
+                List<SpaceLinkDisplay> spaceLinks = spaceToSpaceLinksMap.get(space.getId());
+                List<SpaceLinkDisplay> spaceToSpaceLinksList = new ArrayList<>();
+                if (spaceLinks != null) {
+                    spaceToSpaceLinksList.addAll(spaceLinks);
+                }
+                List<String> spaceAllLink = new ArrayList<>();
+
+                if (spaceToModulelinksList != null) {
+                    spaceToModulelinksList.forEach(link -> {
+                        if (link.getLink() != null && link.getLink().getModule() != null
+                                && link.getLink().getModule().getId() != null) {
+                            spaceAllLink.add(link.getLink().getModule().getId());
+                        }
+                    });
+                }
+                if (spaceToSpaceLinksList != null) {
+                    spaceToSpaceLinksList.forEach(link -> {
+                        if (link.getLink() != null && link.getLink().getTarget() != null
+                                && link.getLink().getTarget().getId() != null) {
+                            spaceAllLink.add(link.getLink().getTarget().getId());
+                        }
+                    });
+                }
+                if (space.getId() != null) {
+                    spaceLinkMap.put(space.getId(), spaceAllLink);
+                }
+            }
+        }
+    }
+
+    /**
+     * creating json for space and module node in the spaceoverview graph
+     * 
+     * @param contextPath This variable holds the contextpath of the application
+     * @param nodeList    List of space/module
+     * @param mapper      For creating ObjectNode
+     * @param nodeArray   Creating Array of ObjectNode for nodes in spaceoverview
+     *                    graph
+     */
+    @Override
+    public void createNodeForOverviewGraph(String contextPath, Iterable<? extends IVSpaceElement> nodeList,
```

Comment:
all methods related to json should be in their own class and called after the space to space/module map has been created. what if we exchange the front end and won't need json anymore but a different format or we need the space to space/module map for something else? The way it is right now the code can't be reused for other purposes.